main.py

```diff
@@ -55,95 +54,81 @@ def main():
-                    print(ffmpeg_call)
-                    subprocess.call(ffmpeg_call)
-
-            # If the user selects to trim...
-            elif to_trim:
-                # Get the start and end trim in points from the user.
-                in_point = set_timestamp(
-                    "Please specify the trim 'in' point")
-                out_point = set_timestamp(
-                    "Please specify the trim 'out' point")
-
-                if to_watermark:
-                    # Make the ffmpeg subprocess call.
-                    ffmpeg_call = [
-                        'ffmpeg', '-ss', in_point,
-                        '-to', out_point,
-                        '-i', sys.argv[1],
-                        '-i', watermark_path(aspect_ratio),
-                        '-filter_complex', "[0]yadif=0:-1:1[v];[v][1]overlay",
-                        '-c:v', 'libx264',
-                        '-pix_fmt', 'yuv420p',
-                        '-metadata', 'copyright=Media Archive for Central England',
-                        '-metadata', 'comment=Please contact MACE to license this footage on 01522 837750', 
-                        '-c:a', 'aac',                       
-                        '-report', output_path
-                    ]
-                    print(ffmpeg_call)
-                    subprocess.call(ffmpeg_call)
-                else:
-                    # Make the ffmpeg subprocess call.
-                    ffmpeg_call = [
-                        'ffmpeg', '-ss', in_point,
-                        '-to', out_point,
-                        '-i', input_file,
-                        '-c:v', 'libx264',
-                        '-pix_fmt', 'yuv420p',
-                        '-vf', 'yadif',
-                        '-metadata', 'copyright=Media Archive for Central England',
-                        '-metadata', 'comment=Please contact MACE to license this footage on 01522 837750',                        
-                        '-c:a', 'aac',                        
-                        '-report', output_path
-                    ]
-                    print(ffmpeg_call)
-                    subprocess.call(ffmpeg_call)
-                    
+            ffmpeg_command = create_ffmpeg_command(input_file)
+
+            print("Transcoding with:", " ".join(ffmpeg_command), "\n")
+
+            subprocess.call(ffmpeg_command)
+
+
+def create_ffmpeg_command(input_file):
+    output_path = set_output_path(input_file)
+
+    to_watermark = set_watermark_status()
+    to_trim = set_trim_status()
+
+    aspect_ratio = stream_aspect_ratio(input_file)
+    dar = display_aspect_ratio(aspect_ratio)
+    watermark_file = watermark_path(aspect_ratio)
+
+    ffmpeg_program_call = [
+        "ffmpeg"
+    ]
+
+    input_video_file = [
+        "-i", input_file
+    ]
+
+    watermark_settings = [
+        "-i", watermark_file,
+        "-filter_complex", "[0]yadif=0:-1:1[v];[v][1]overlay"  # overlay filter for supplied png
```

Comment:
Ah sorry, just realised that I copied the `yadif` in here by accident. Will fix later on!

Reply:
Opps - just saw some of the Twitter discussion. It looks like you can't have `-filter_complex` AND `-vf`? Another thing for me to fix ...

Reply:
Changed to simply `overlay` in 5b2e721 :smile: